var/spack/repos/builtin/packages/libmad/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libmad(AutotoolsPackage):
+    """MAD is a high-quality MPEG audio decoder."""
+
+    homepage = "https://www.underbit.com/products/mad/"
+    url      = "https://download.sourceforge.net/project/mad/libmad/0.15.1b/libmad-0.15.1b.tar.gz"
+
+    version('0.15.1b', sha256='bbfac3ed6bfbc2823d3775ebb931087371e142bb0e9bb1bee51a76a6e0078690')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('m4', type='build')
+
+    patch('libmad-0.15.1b.patch')
+
+    force_autoreconf = True
```

Comment:
One thing I often do is try to manually patch and autoreconf myself, and then see if I can patch `configure` instead. This has the advantage of not adding 4 more dependencies.

Reply:
I tried this after your mention, but the patch became quite messy and hard to read. Personally I would prefer easily readable patch with more requirements/steps than complicated patch with less requirements/steps.

I used basically
```sh
wget https://download.sourceforge.net/project/mad/libmad/0.15.1b/libmad-0.15.1b.tar.gz
tar xf libmad-0.15.1b.tar.gz
cp -r libmad-0.15.1b libmad-0.15.1b.orig
patch -p0 < libmad-0.15.1b.patch
cd libmad-0.15.1b
autoreconf -fi
cd ..
diff -Naur libmad-0.15.1b.orig/configure libmad-0.15.1b/configure > libmad-0.15.1b-configure.patch
```

Reply:
Just to clarify: with autoconf 2.69 the patch is over 34000 lines. The original configure is from 2004 and created with autoconf 2.59. With autoconf 2.59 the `autoreconf`-command did not work.